forum/src/main/java/unipd/se18/ocrcamera/forum/viewmodels/AddPost_VM.java

```diff
@@ -0,0 +1,320 @@
+
+    /**
+     * Keys of the JSON strings value for a post
+     * @see <a href="https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html">Enum Types</a>
+     */
+    public enum JSONPostKey {
+        ID("ID"),
+        TITLE("title"),
+        MESSAGE("message"),
+        DATE("date"),
+        LIKES("likes"),
+        COMMENTS("comments"),
+        AUTHOR("author");
+
+        public String value;
+
+        /**
+         * Defines an object of type JSONPostKey
+         * @param value The value of the key for the JSON format of the post
+         */
+        JSONPostKey(String value){ this.value = value; }
+    }
+
+    /*
+    The next listener is provided by the class used for communicating with the server that hosts
+    the forum (RequestManager).
+     */ 
+
+    /**
+     * Listener useful to receive notification from the requests manager
+     * More details at: {@link RequestManager.RequestManagerListener}
+     */
+    private RequestManager.RequestManagerListener requestManagerListener =
+            new RequestManager.RequestManagerListener() {
+        /**
+         * Notifies that a post was added correctly
+         * @param response The network request's response
+         */
+        @Override
+        public void onRequestFinished(String response) {
+            // Post added
+            Log.d(TAG,"onRequestFinished -> response: " + response);
+	    // Notifies the event to the View
+            notifier.onPostAdded(response);
+        }
+
+        /**
+         * Notifies a connection problem to the network
+         * @param message The error message
+         */
+        @Override
+        public void onConnectionFailed(String message) {
+            // Connection problem
+            Log.d(TAG,"onConnectionFailed -> message: " + message);
+	    // Notifies the event to the View
+            notifier.onConnectionFailed(message);
+        }
+
+        /**
+         * Notifies a failure of a sending parameters process addressed to the server
+         * @param message The error message
+         */
+        @Override
+        public void onParametersSendingFailed(String message) {
+            // Parameters not sent correctly
+            Log.d(TAG,"onParametersSendingFailed -> message: " + message);
+	    // Notifies the event to the View
+            notifier.onParametersSendingFailed(message);
+        }
+    };
+
+    /**
+     * Adds a post to the forum
+     * More details at: {@link RequestManager},
+     * {@link RequestManager#sendRequest(Context, ArrayList)}.
+     * @param context The reference of the activity/fragment that calls this method
+     * @param title The new post's title
+     * @param message The new post's message
+     * @param author The new post's author
+     * @author Pietro Prandini (g2)
+     */
+    @Override
+    public void addPostToForum(final Context context, String title, String message, String author) {
+        Log.i(TAG,"addPostToForum");
```

Comment:
In my experience the logs don't compromise the user experience and at the same time they support the developers, so I usually left them but there are some levels of logs described also in the [android API](https://developer.android.com/reference/android/util/Log) and the "verbose" level it could be considered only for using under development.